dll/win32/shell32/CNewMenu.cpp

```diff
@@ -406,38 +406,51 @@ CNewMenu::SHELLNEW_ITEM *CNewMenu::FindItemFromIdOffset(UINT IdOffset)
 
 HRESULT CNewMenu::SelectNewItem(LONG wEventId, UINT uFlags, LPWSTR pszName, BOOL bRename)
 {
-    CComPtr<IShellBrowser> lpSB;
+    CComPtr<IShellFolder> lpDesktopSF;
+    CComPtr<IShellFolder> lpCurrentSF;
     CComPtr<IShellView> lpSV;
     HRESULT hr = E_FAIL;
-    LPITEMIDLIST pidl;
-    PITEMID_CHILD pidlNewItem;
+    LPITEMIDLIST pidlFullNewItem = NULL;
+    PIDLIST_ABSOLUTE pidlNewItem = NULL;
```

Comment:
If you use an `CComHeapPtr<ITEMIDLIST_ABSOLUTE>` you do not need to free it manually

Reply:
I got to know something new. Do I need to change the code?

sorry for late reply. my reply was in pending mode.